### DISCUSSION TITLE
LINE oauth error

### DISCUSSION THREAD

**Discussion by misway** [Hebilicious/authjs-nuxt#81] [Q&A] [answered]
When attempting to use the LINE provider and sending an authentication request, an error occurs on the server, and the LINE login screen does not appear, regardless of whether it's in the dev or CFP.

`[auth][error][OAuthCallbackError]: OAuth Provider returned an error. Read more at https://errors.authjs.dev#oauthcallbackerror`

Entering the authorization URL in the browser, the login screen appears smoothly. 

`https://access.line.me/oauth2/v2.1/authorize?client_id={myid}&redirect_uri=http://localhost:3000/api/auth/callback/line&response_type=code&scope=openid%20email%20profile&state=abcdef`

Upon successful login, it redirects to the URL that was set as the redirect_uri localhost. At this point, the authjs in the localhost dev correctly receives the completed authentication information and displays it.

Therefore, I speculate that the issue might lie in the first part of the process. Has anyone tested with LINE or can provide suggestions?

**Comment by Hebilicious** [comment, marked as answer]
> When attempting to use the LINE provider and sending an authentication request, an error occurs on the server, and the LINE login screen does not appear, regardless of whether it's in the dev or CFP.
> 
> `[auth][error][OAuthCallbackError]: OAuth Provider returned an error. Read more at https://errors.authjs.dev#oauthcallbackerror`
> 
> Entering the authorization URL in the browser, the login screen appears smoothly.
> 
> `https://access.line.me/oauth2/v2.1/authorize?client_id={myid}&redirect_uri=http://localhost:3000/api/auth/callback/line&response_type=code&scope=openid%20email%20profile&state=abcdef`
> 
> Upon successful login, it redirects to the URL that was set as the redirect_uri localhost. At this point, the authjs in the localhost dev correctly receives the completed authentication information and displays it.
> 
> Therefore, I speculate that the issue might lie in the first part of the process. Has anyone tested with LINE or can provide suggestions?

This might be related to auth.js and not to this module. To debug this, it's probably easier to fork this repo and try the line provider in the playground. It will be easier to debug what's happening with the sign-in method.
